tests/resources/test_table_config.py

```diff
@@ -319,6 +315,81 @@ def test_default_for_material_failure(collection: TableConfigCollection):
         )
 
 
+def test_from_query(collection: TableConfigCollection, session):
+    """Test that default for material hits the right route"""
+    query = GemdQueryDataFactory()
+    config = TableConfigDataFactory()
+
+    config_resp = {
+        'config': config,
+        'ambiguous': [
+            [
+                TerminalMaterialIdentifier(name='foo', headers=['foo'], scope='id').dump(),
+                IdentityColumn(data_source='foo').dump(),
+            ]
+        ],
+    }
+    session.responses.append(config_resp)
+    fake_call = FakeCall(
+        method='POST',
+        path=f'teams/{collection.team_id}/table-configs/from-query',
+        params={
+            'name': config['name'],
+            'description': config['description'],
+            'algorithm': TableFromGemdQueryAlgorithm.MULTISTEP_MATERIALS,
+        },
+        json=query,
+    )
+
+    collection.from_query(
+        name=config['name'],
+        description=config['description'],
+        gemd_query=GemdQuery.build(query),
+        algorithm=TableFromGemdQueryAlgorithm.MULTISTEP_MATERIALS
+    )
+    assert 1 == session.num_calls
+    assert session.last_call.method == fake_call.method
+    assert session.last_call.path == fake_call.path
+    assert session.last_call.params == fake_call.params
+    last_query: GemdQuery = GemdQuery.build(session.last_call.json)
+    fake_query: GemdQuery = GemdQuery.build(fake_call.json)
+    assert last_query.datasets == fake_query.datasets
+    assert last_query.object_types == fake_query.object_types
+    assert last_query.schema_version == fake_query.schema_version
+    assert last_query.dump() == fake_query.dump()  # Ordering issues
+
+
+def test_from_nameless_query_and_register(collection: TableConfigCollection, session):
+    """Test that default for material hits the right route"""
+    query = GemdQueryDataFactory()
+    config = TableConfigDataFactory(generation_algorithm=TableFromGemdQueryAlgorithm.MULTISTEP_MATERIALS)
+
+    dataset_resps = [DatasetDataFactory(id=dataset) for dataset in query['datasets']]
+    config_resp = {
+        'config': config,
+        'ambiguous': [
+            [
+                TerminalMaterialIdentifier(name='foo', headers=['foo'], scope='id').dump(),
+                IdentityColumn(data_source='foo').dump(),
+            ]
+        ],
+    }
+    register_resp = TableConfigResponseDataFactory(version__ara_definition=config)
+
+    session.responses.extend(dataset_resps)
+    session.responses.append(config_resp)
+    session.responses.append(register_resp)
+
+    generated, _ = collection.from_query(
+        gemd_query=GemdQuery.build(query),
+        description='my_description',
+        register_config=True
+    )
+    assert session.num_calls == len(query['datasets']) + 1 + 1
+    assert generated != TableConfig.build(config)  # Because it has ids
```

Comment:
Is there any code change that can be introduced to falsify this assertion? I'm just curious about why you are validating it. It might be obvious but I'm not super familiar with this.

Reply:
The table config returned by the registration route augments the table config returned by the from_query route by adding registration metadata to the object (id, version number...).  This is making sure that when you flip the `register_config` flag, you get the return of the register route, not the from_query route.

In theory, this test could be rewritten as `The table config has meta data, and if I ignore it, then they are the same` but I don't think that adds meaningful additional coverage.